core/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -114,6 +116,18 @@ public ParseFieldMatcher parseFieldMatcher() {
         return parseFieldMatcher;
     }
 
+    public void setTask(SearchTask task) {
```

Comment:
It'd be nice to have javadoc for these, especially given that task changes for scrolls.